File: Exercises/Exercise04/java_dgemm/src/benchmark/Reader.java
package benchmark;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;

public class Reader {
    private String valueType;
    private int matrixSize;
    private String alpha;
    private String beta;
    private String[][] matrixA;
    private String[][] matrixB;
    private String[][] matrixC;
    private File matrixCFile;

    public String getValueType(){ return valueType; }
    public int getMatrixSize(){ return matrixSize; }
    public String getAlpha(){ return alpha; }
    public String getBeta(){ return beta; }
    public String[][] getMatrixA(){ return matrixA; }
    public String[][] getMatrixB(){ return matrixB; }
    public String[][] getMatrixC(){ return matrixC; }
    public File getMatrixCFile(){ return matrixCFile; }

    public void read(){
        File valueTypeFile = new File("..\\Data\\value_type.txt");
        File matrixSizeFile = new File("..\\Data\\matrix_size.txt");
        File alphaFile = new File("..\\Data\\alpha.txt");
        File betaFile = new File("..\\Data\\beta.txt");
        File matrixAFile = new File("..\\Data\\matrix_a.txt");
        File matrixBFile = new File("..\\Data\\matrix_b.txt");
        matrixCFile = new File("..\\Data\\matrix_c.txt");

        try (BufferedReader valueTypeReader = new BufferedReader(new FileReader(valueTypeFile))) {
            valueType = valueTypeReader.readLine();
        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        }

        try (BufferedReader matrixSizeReader = new BufferedReader(new FileReader(matrixSizeFile))) {
            matrixSize = Integer.parseInt(matrixSizeReader.readLine());
        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        } catch (NumberFormatException e) {
            System.err.println("Error: " + e.getMessage());
        }

        matrixA = new String[matrixSize][matrixSize];
        matrixB = new String[matrixSize][matrixSize];
        matrixC = new String[matrixSize][matrixSize];

        try (BufferedReader alphaReader = new BufferedReader(new FileReader(alphaFile))) {
            alpha = alphaReader.readLine();
        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        }

        try (BufferedReader betaReader = new BufferedReader(new FileReader(betaFile))) {
            beta = betaReader.readLine();
        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        }

        try (BufferedReader matrixAReader = new BufferedReader(new FileReader(matrixAFile))) {
            String input;
            int counter = 0;
            while ((input = matrixAReader.readLine()) != null){
                matrixA[counter] = input.split(" ");
                counter++;
            }
        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        }

        try (BufferedReader matrixBReader = new BufferedReader(new FileReader(matrixBFile))) {
            String input;
            int counter = 0;
            while ((input = matrixBReader.readLine()) != null){
                matrixB[counter] = input.split(" ");
                counter++;
            }
        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        }

        try (BufferedReader matrixCReader = new BufferedReader(new FileReader(matrixCFile))) {
            String input;
            int counter = 0;
            while ((input = matrixCReader.readLine()) != null){
                matrixC[counter] = input.split(" ");
                counter++;
            }
        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
}

File: Exercises/Exercise04/Csharp.Dgemm/Csharp.Dgemm/Reader.cs
using System;
using System.IO;

namespace Csharp.Dgemm
{
    public class Reader
    {
        private string valueType;
        private int matrixSize;
        private string alpha;
        private string beta;
        private string[,] matrixA;
        private string[,] matrixB;
        private string[,] matrixC;
        private string matrixCPath;

        public string ValueType { get { return valueType; } }
        public int MatrixSize { get { return matrixSize;} }
        public string Alpha { get { return alpha; } }
        public string Beta { get { return beta;} }
        public string[,] MatrixA { get { return matrixA;} }
        public string[,] MatrixB { get { return matrixB; } }
        public string[,] MatrixC { get { return matrixC; } }
        public string MatrixCPath { get { return matrixCPath; } }

        public void Read()
        {
            try
            {
                string valueTypePath = "..\\..\\..\\..\\Data\\value_type.txt";
                string matrixSizePath = "..\\..\\..\\..\\Data\\matrix_size.txt";
                string alphaPath = "..\\..\\..\\..\\Data\\alpha.txt";
                string betaPath = "..\\..\\..\\..\\Data\\beta.txt";
                string matrixAPath = "..\\..\\..\\..\\Data\\matrix_a.txt";
                string matrixBPath = "..\\..\\..\\..\\Data\\matrix_b.txt";
                matrixCPath = "..\\..\\..\\..\\Data\\matrix_c.txt";

                using(StreamReader sr = new StreamReader(valueTypePath))
                {
                    valueType = sr.ReadLine();
                }

                using (StreamReader sr = new StreamReader(matrixSizePath))
                {
                    matrixSize = Int32.Parse(sr.ReadLine());
                }

                matrixA = new string[matrixSize, matrixSize];
                matrixB = new string[matrixSize, matrixSize];
                matrixC = new string[matrixSize, matrixSize];

                using (StreamReader sr = new StreamReader(alphaPath))
                {
                    alpha = sr.ReadLine();
                }

                using (StreamReader sr = new StreamReader(betaPath))
                {
                    beta = sr.ReadLine();
                }

                using (StreamReader sr = new StreamReader(matrixAPath))
                {
                    string line;
                    int i = 0;
                    while ((line = sr.ReadLine()) != null)
                    {
                        string[] container = line.Split(' ');
                        int j = 0;
                        foreach (string s in container)
                        {
                            matrixA[i, j] = s;
                            j++;
                        }
                        i++;
                    }
                }

                using (StreamReader sr = new StreamReader(matrixBPath))
                {
                    string line;
                    int i = 0;
                    while ((line = sr.ReadLine()) != null)
                    {
                        string[] container = line.Split(' ');
                        int j = 0;
                        foreach (string s in container)
                        {
                            matrixB[i, j] = s;
                            j++;
                        }
                        i++;
                    }
                }

                using (StreamReader sr = new StreamReader(matrixCPath))
                {
                    string line;
                    int i = 0;
                    while ((line = sr.ReadLine()) != null)
                    {
                        string[] container = line.Split(' ');
                        int j = 0;
                        foreach (string s in container)
                        {
                            matrixC[i, j] = s;
                            j++;
                        }
                        i++;
                    }
                }
            }
            catch (IOException e)
            {
                Console.WriteLine(e.Message);
            }
            catch (InvalidCastException e)
            {
                Console.WriteLine(e.Message);
            }
        }
    }
}
